src/js/pages/nodes/NodeDetailPage.js

```diff
@@ -0,0 +1,373 @@
+
+    this.store_listeners = [
+      {name: 'summary', events: ['success']},
+      {name: 'state', events: ['success']},
+      {name: 'nodeHealth', events: ['nodeSuccess', 'nodeError', 'unitsSuccess', 'unitsError']}
+    ];
+
+    this.tabs_tabs = {
+      tasks: 'Tasks',
+      health: 'Health',
+      details: 'Details'
+    };
+
+    this.state = {
+      currentTab: Object.keys(this.tabs_tabs).shift()
+    };
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    }, this);
+  }
+
+  componentDidMount() {
+    super.componentDidMount(...arguments);
+
+    this.mountedAt = Date.now();
+
+    let node = this.getNode();
+    this.internalStorage_update({node});
+
+    if (node) {
+      NodeHealthStore.fetchNodeUnits(node.hostname);
+    }
+  }
+
+  componentWillUpdate() {
+    super.componentWillUpdate(...arguments);
+
+    if (!this.internalStorage_get().node) {
+      this.internalStorage_update({node: this.getNode()})
+    }
+  }
+
+  getBasicInfo(node) {
+    return (
+      <div className="detail-page-header">
+        <h1 className="inverse flush">
+          {node.hostname}
+        </h1>
+        {this.getSubHeader(node)}
+      </div>
+    );
+  }
+
+  getBreadcrumbs(nodeID) {
+    return (
+      <h5 className="inverse">
+        <Link className="headline emphasize" to="nodes">
+          Nodes
+        </Link>
+        <span style={{'margin': '0 15px'}}>
+          ›
+        </span>
+        {nodeID}
+      </h5>
+    );
+  }
+
+  getCharts(itemType, item) {
+    if (!item) {
+      return null;
+    }
+
+    let states = MesosSummaryStore.get('states');
+    let resources = states[`getResourceStatesFor${itemType}IDs`]([item.id]);
+
+    let charts = [
+      this.getResourceChart('cpus', resources),
+      this.getResourceChart('mem', resources),
+      this.getResourceChart('disk', resources)
+    ];
+
+    return charts.map(function (chart, i) {
+      return (
+        <div key={i} className="column-12 column-mini-4">
+          <div className="row chart">
+            {chart}
+          </div>
+        </div>
+      );
+    });
+  }
+
+  getLoadingScreen() {
+    return (
+      <div className="container container-fluid container-pod text-align-center
+        vertical-center inverse">
+        <div className="row">
+          <div className="ball-scale">
+            <div />
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  getNode() {
+    return CompositeState.getNodesList().filter(
+      {ids: [this.props.params.nodeID]}
+    ).last();
+  }
+
+  getNotFound() {
+    return (
+      <div className="container container-fluid container-pod text-align-center">
+        <h3 className="flush-top text-align-center">
+          Error finding node
+        </h3>
+        <p className="flush">
+          {`Did not find a node by the id "${this.props.params.nodeID}"`}
+        </p>
+      </div>
+    );
+  }
+
+  getResourceChart(resource, totalResources) {
```

Comment:
This seems copy-pasted from `SidePanelContents.js`, could we make it a component, since we will need this for both services and nodes?